internal/clients/substrate/interface.go

```diff
@@ -0,0 +1,13 @@
+package substrate
```

Comment:
Move the content of this file into substrate.go (and probably no need for a subpackage substrate, it can live in the clients just fine)

Reply:
I moved the content but left the subpackage for now. will revisit after finalizing the PR